app-android/src/main/java/com/ustadmobile/port/android/view/ClazzEditFragment.kt

```diff
@@ -284,4 +317,376 @@ class ClazzEditFragment() : UstadEditFragment<ClazzWithHolidayCalendarAndSchoolA
+                value = uiState.entity?.clazzStartTime ?: 0,
+                label = stringResource(id = R.string.start_date),
+                error = uiState.clazzStartDateError,
+                enabled = uiState.fieldsEnabled,
+                timeZoneId = uiState.entity?.clazzTimeZone ?: "UTC",
+                onValueChange = {
+                    onClazzChanged(
+                        uiState.entity?.shallowCopy {
+                            clazzStartTime = it
+                        }
+                    )
+                }
+            )
+        }
+
+        item {
+            UstadDateEditTextField(
+                value = uiState.entity?.clazzEndTime ?: 0,
+                label = stringResource(id = R.string.end_date).addOptionalSuffix(),
+                error = uiState.clazzEndDateError,
+                enabled = uiState.fieldsEnabled,
+                timeZoneId = uiState.entity?.clazzTimeZone ?: "UTC",
+                onValueChange = {
+                    onClazzChanged(
+                        uiState.entity?.shallowCopy {
+                            clazzEndTime = it
+                        }
+                    )
+                }
+            )
+        }
+
+        item {
+            UstadTextEditField(
+                value = uiState.entity?.clazzTimeZone ?: "",
+                label = stringResource(id = R.string.timezone),
+                enabled = uiState.fieldsEnabled,
+                onClick = { onClickTimezone() },
+                onValueChange = {}
+            )
+        }
+
+        item {
+            Spacer(modifier = Modifier.height(20.dp))
+        }
+
+        item {
+            Text(text = stringResource(id = R.string.course_blocks))
+        }
+
+        item {
+            ListItem(
+                modifier = Modifier.clickable {
+                    onClickAddCourseBlock()
+                },
+                text = { Text(stringResource(id = R.string.add_block)) },
+            )
+        }
+
+        items (
+            items = uiState.courseBlockList
+        ) { courseBlock ->
+
+            val CONTENT_ENTRY_TYPE_ICON_MAP = mapOf(
+                ContentEntry.TYPE_EBOOK to R.drawable.ic_book_black_24dp,
+                ContentEntry.TYPE_VIDEO to R.drawable.video_youtube,
+                ContentEntry.TYPE_DOCUMENT to R.drawable.text_doc_24px,
+                ContentEntry.TYPE_ARTICLE to R.drawable.article_24px,
+                ContentEntry.TYPE_COLLECTION to R.drawable.collections_24px,
+                ContentEntry.TYPE_INTERACTIVE_EXERCISE to R.drawable.ic_baseline_touch_app_24,
+                ContentEntry.TYPE_AUDIO to R.drawable.ic_audiotrack_24px
+            )
+            val image = if(courseBlock.cbType == CourseBlock.BLOCK_CONTENT_TYPE)
+                courseBlock.entry?.contentTypeFlag
+            else
+                courseBlock.cbType
+
+            val courseBlockEditAlpha: Float = if (courseBlock.cbHidden) 0.5f else 1f
+
+            ListItem(
```

Comment:
To implement the indent: the start padding of the ListItem should be 8dp * the indent level of the block.